### PR TITLE
fix: csv to oscal cd task

### PR DESCRIPTION
**Pull request by degenaro**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [ ] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

Consume parameter related columns from csv file and produce corresponding entries in component-definition.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: trestle/tasks/csv_helper.py [hunk 1]
```diff
@@ -28,22 +28,41 @@ class Column():
...
-            'Parameter_Description',
-            'Parameter_Default_Value',
-            'Parameter_Value_Alternatives'
         ]
-        self.help_list = []
-        for column in self.columns:
-            self.help_list.append(column)
+        self._columns_optional = [
+            'Parameter_Id', 'Parameter_Description', 'Parameter_Default_Value', 'Parameter_Value_Alternatives'
+        ]
```
Comment (vikas-agarwal76) [inline trestle/tasks/csv_helper.py:42]: There are other optional columns also that we decided earlier like fetchers, checkers, etc. Please see the best practices spreadsheet to CSV file that you generated earlier.

File: trestle/tasks/csv_helper.py [hunk 2]
```diff
@@ -28,22 +28,41 @@ class Column():
...
+        self.help_list_required = []
+        for column in self._columns_required:
+            self.help_list_required.append(column)
+        self.help_list_optional = []
+        for column in self._columns_optional:
+            self.help_list_optional.append(column)
+
+        logger.info(f'init: {self._columns_required}')
+
+    def is_user_column(self, column_name) -> bool:
```
Comment (enikonovad) [inline trestle/tasks/csv_helper.py:52]: please add typing

**Review by enikonovad — APPROVED** [review]
LGTM - from the code perspective

File: trestle/tasks/csv_helper.py (hunk 1 above)
Comment (degenaro) [reply trestle/tasks/csv_helper.py:50]: Fixed.
optional columns:   Parameter_Id
                      Parameter_Description
                      Parameter_Default_Value
                      Parameter_Value_Alternatives
                      Check_Id
                      Check_Description
                      Fetcher
                      Fetcher_Description

**Pull request merged**